File: api/src/main/java/com/juliopredictor/api/Dashboard/Predictor/Infrastructure/Gateway/CoinMarketCap/responseDto/CoinMarketCapListTop300Response.java
package com.juliopredictor.api.Dashboard.Predictor.Infrastructure.Gateway.CoinMarketCap.responseDto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CoinMarketCapListTop300Response {
    private CoinMarketCapStatusResponse status;
    private List<CoinMarketCapMapEntity> data;
}

File: webapp/src/models/CoinMarketCapListTop300Response.ts
import { CoinMarketCapMapEntity } from "./CoinMarketCapMapEntity";
import { CoinMarketCapStatusResponse } from "./CoinMarketCapStatusResponse";

export interface CoinMarketCapListTop300Response{
    status : CoinMarketCapStatusResponse,
    data : Array<CoinMarketCapMapEntity>,
}